apps/system/js/app_chrome.js

```diff
@@ -410,63 +417,88 @@
                         this.app.themeColor, true);
       }
     }, () => {
-      console.log('Failed to pin ' + this.app.config.url);
+      console.error('Failed to pin ' + this.app.config.url);
     });
   };
 
+  AppChrome.prototype.getSiteUrl = function() {
+    var pageUrl = this.app.config.url;
+    var webManifest = this.app.webManifest;
+    var startURL = (webManifest && this.app.webManifest.start_url) ?
+      new URL(this.app.webManifest.start_url, pageUrl) : null;
+    return startURL ? startURL.href : pageUrl;
+  };
+
   /**
    * Pin current site in bookmarks database.
    */
   AppChrome.prototype.pinSite = function ac_pinSite() {
-    this.pin();
-
-    LazyLoader.load('shared/js/url_helper.js').then(() => {
-      var siteObject = {};
-      var manifestURL = this.app.webManifestURL;
-      var manifestObject = this.app.webManifest;
-      var pageUrl = this.app.config.url;
-      var systemBanner = this.systemBanner;
-
-      siteObject.type = 'url';
-      siteObject.iconable = false;
-      siteObject.icons = this.app.favicons;
-      siteObject.frecency = 1;
-      siteObject.pinned = true;
-      siteObject.pinnedFrom = pageUrl;
-
-      if (manifestURL && manifestObject) {
-        var startUrl =
-        UrlHelper.resolveUrl(manifestObject.start_url, pageUrl) || pageUrl;
-        siteObject.id = startUrl;
-        siteObject.url = startUrl;
-        siteObject.webManifestUrl = manifestURL;
-        siteObject.webManifest = manifestObject;
-        siteObject.name = manifestObject.short_name || manifestObject.name ||
-          UrlHelper.getHostname(pageUrl);
-        siteObject.scope =
-          UrlHelper.resolveUrl(manifestObject.scope, pageUrl) ||
-          UrlHelper.resolveUrl('/', pageUrl);
+    var siteObject = {};
+    var manifestURL = this.app.webManifestURL;
+    var manifestObject = this.app.webManifest;
+    var pageURL = new URL(this.app.config.url);
+    var scope = pageURL.origin + '/';
```

Comment:
Yes, because otherwise a website located at `http://example.industries` would share the same origin as `http://example.in`. The `/` prevents that from happening.